packages/engine-server/src/util/impl/inMemoryNoteCacheImpl.ts

```diff
@@ -0,0 +1,47 @@
+import { DendronError, NoteProps } from "@dendronhq/common-all";
+import { InMemoryNoteCache } from "../inMemoryNoteCache";
+
+export class InMemoryNoteCacheImpl implements InMemoryNoteCache {
+  /** Note map which maps lowercase file name to a list {@link NoteProps}
+   *  that have the matching file name */
+  private readonly mapFNameToNotes: Map<string, NoteProps[]>;
+
+  static createCache(notes: NoteProps[]) {
+    return new InMemoryNoteCacheImpl(
+      InMemoryNoteCacheImpl.initializeFileNameMap(notes)
+    );
+  }
+
+  private constructor(noteMap: Map<string, NoteProps[]>) {
+    this.mapFNameToNotes = noteMap;
+  }
+
+  private static initializeFileNameMap(notes: NoteProps[]) {
+    const map = new Map<string, NoteProps[]>();
```

Comment:
Is the reason this is `NoteProps[]` and not `NoteProps` as the value because multiple vaults may have the same note name?